Examples/Algorithms/Vertexing/src/AdaptiveMultiVertexFinderAlgorithm.cpp

```diff
@@ -181,12 +261,29 @@ ActsExamples::AdaptiveMultiVertexFinderAlgorithm::execute(
     }
   }
 
-  //////////////////////////////////////////////
-  /* Full tutorial example code for reference */
-  //////////////////////////////////////////////
+  const Acts::AdaptiveMultiVertexFinder* vertexFinder = &m_vertexFinder;
+
+  // Temporary vertex finder for the truth seeder
+  std::optional<Acts::AdaptiveMultiVertexFinder> temporaryVertexFinder;
+  if (m_cfg.seedFinder == SeedFinder::TruthSeeder) {
+    // In case of the truth seeder, we need to wire the truth vertices into the
+    // vertex finder
+
+    const auto& truthParticles = m_inputTruthParticles(ctx);
+    const auto& truthVertices = m_inputTruthVertices(ctx);
+
+    // Build a new vertex seeder with the truth vertices
+    auto vertexSeeder =
+        makeTruthVertexSeeder(m_cfg, truthParticles, truthVertices, logger());
```

Comment:
You have to recreate this because it needs to contain a list of truth vertices right? 

I've been thinking about this, and I guess one idea could be to use the state to pipe this through. The generic `makeState` function does not have access to the event context, but an algorithm-level function could conditionally downcast to the truth vertex seeder state and inject the truth vertices. 

Do you think that would be cleaner?

Reply:
That would have been my favorite solution as well but the seeder state is created inside the `find` function
https://github.com/acts-project/acts/blob/9b51fed8be0ef56416bff99583751adb733c1191/Core/src/Vertexing/AdaptiveMultiVertexFinder.cpp#L35
So we would have to pipe it into `find` or change the way the seeder state is dealt with

Reply:
I see. Maybe we need a way for the AMVF to externally initialize the state then?